src/main/java/spoon/decompiler/SpoonClassFileTransformer.java

```diff
@@ -0,0 +1,169 @@
+/**
+ * Copyright (C) 2006-2018 INRIA and contributors
+ * Spoon - http://spoon.gforge.inria.fr/
+ *
+ * This software is governed by the CeCILL-C License under French law and
+ * abiding by the rules of distribution of free software. You can use, modify
+ * and/or redistribute the software under the terms of the CeCILL-C license as
+ * circulated by CEA, CNRS and INRIA at http://www.cecill.info.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the CeCILL-C License for more details.
+ *
+ * The fact that you are presently reading this means that you have had
+ * knowledge of the CeCILL-C license and that you accept its terms.
+ */
+package spoon.decompiler;
+
+import org.benf.cfr.reader.Main;
+import spoon.IncrementalLauncher;
+import spoon.SpoonModelBuilder;
+import spoon.reflect.CtModel;
+import spoon.reflect.declaration.CtType;
+import spoon.support.Experimental;
+
+import java.io.File;
+import java.io.IOException;
+import java.lang.instrument.ClassFileTransformer;
+import java.lang.instrument.IllegalClassFormatException;
+import java.nio.file.Files;
+import java.security.ProtectionDomain;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.function.Predicate;
+
+@Experimental
+public abstract class SpoonClassFileTransformer implements ClassFileTransformer {
+
+	Predicate<String> classNameFilter;
+	String pathToDecompiled;
+	String pathToRecompile;
+	String pathToCache;
+	File cache;
+	Set<String> classPath;
+	public Decompiler decompiler;
+	Set<File> inputSources;
```

Comment:
I guess all the members above should be private or at least protected. WDYT?

Reply:
Yes definitly.